testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java

```diff
@@ -92,14 +91,16 @@
 
     @Test
     public void test(
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_1) URL baseURL1,
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_2) URL baseURL2)
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_1) URL baseURLDep1,
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_2) URL baseURLDep2)
             throws URISyntaxException, IOException {
+        URL baseURL1 = new URL(baseURLDep1.toExternalForm() + "/");
```

Comment:
I'd prefer that we add SimpleServlet.createURI(...) methods that accept a URI which should avoid the need for the convoluted URL -> string -> URL -> string -> URL conversions.